onnxscript/rewriter/llama_rule_sets.py

```diff
@@ -0,0 +1,42 @@
+import onnxscript.rewriter.no_op as no_op
+import onnxscript.rewriter.pattern as orp
+from onnxscript.rewriter import pattern
+
+op = pattern.onnxop
+
+
+def transpose_identity(x):
+    return op.Transpose(x, perm=[0, 1, 2])
```

Comment:
Hi Xavier, you can write this pattern to get the perm-attribute value from input model as below: using this to illustrate how we get values from the model itself:
```py
from __future__ import annotations
import onnxscript.rewriter as rewriter
import onnxscript.rewriter.pattern as orp
import onnxscript.ir as ir
from onnxscript import script, opset17, FLOAT
import onnx.printer

op = orp.onnxop

def transpose(x, perm):
    return op.Transpose(x, perm=perm)

def check_identity_perm_impl(x: ir.Value, perm: ir.Attr | ir.RefAttr):
    # Inside a function, an attribute may be a RefAttr and its value may not be available.
    if isinstance(perm, ir.RefAttr):
        return False
    if perm.type == ir.AttributeType.INTS:
        if perm.value == list(range(len(perm.value))):
            return True
    return False

def check_identity_perm(match_bindings):
    # Current API uses this interface. We would like to switch to the one in "check_identity_perm_impl".
    return check_identity_perm_impl(**match_bindings)

def identity_perm(op, x: ir.Value, perm: ir.Attr | ir.RefAttr):
    return op.Identity(x)

rule = orp.RewriteRule(transpose, identity_perm, check_identity_perm)

rules = orp.RewriteRuleSet([rule])

@script()
def example(X: FLOAT[2, 4, 1024]) -> FLOAT[2, 4, 1024]:
    return opset17.Transpose(X, perm=[0, 1, 2])

model = example.to_model_proto()

optimized = rewriter.rewrite(model, pattern_rewrite_rules=rules)

print(onnx.printer.to_text(optimized))

@script()
def example2(X: FLOAT[2, 1024]) -> FLOAT[2, 1024]:
    return opset17.Transpose(X, perm=[1, 0])

model2 = example2.to_model_proto()

optimized2 = rewriter.rewrite(model2, pattern_rewrite_rules=rules)

print(onnx.printer.to_text(optimized2))

```

Reply:
Adding type annotations to the pattern, it will look like the below (just FYI):
```py
def transpose(x: orp.Var, perm: orp.Var) -> orp.ValuePattern:
    return op.Transpose(x, perm=perm)
```